lighthouse-core/lib/task-groups.js

```diff
@@ -5,9 +5,11 @@
  */
 'use strict';
 
+/** @typedef {'parseHTML'|'styleLayout'|'paintCompositeRender'|'scriptParseCompile'|'scriptEvaluation'|'garbageCollection'|'other'} TaskGroupIds */
```

Comment:
I've been wondering is there not a way to get these programmtically via some `keyof` magic?

Reply:
I know we're going the opposite direction by declaring the type of the object as matching this, but ya know just for knowledge :)

Reply:
> I've been wondering is there not a way to get these programmtically via some keyof magic?

I tried really hard :) but I couldn't find a way to get out of the circularity of needing `TaskGroup.id` to be able to index `taskGroups` in `mainthread-work-breakdown.js`, which means the `taskGroups` declaration needs to be typed, not just inferred (it would go to `string`), which means you can't meaningfully do `keyof` on it

Reply:
actually, you could do one really terrible thing...
```js
const taskGroups = {
  parseHTML: {
    id: /** @type {'parseHTML'} */ ('parseHTML'),
    // ...
  },
  styleLayout: {
    id: /** @type {'styleLayout'} */ ('styleLayout'),
    // ...
  }, // ...
};
```

Reply:
alright thanks good to know!